src/audio/module_adapter/module_adapter.c

```diff
@@ -736,7 +736,8 @@ static int module_adapter_simple_copy(struct comp_dev *dev)
 		src_c = attr_container_of(mod->input_buffers[i].data,
 					  struct comp_buffer __sparse_cache,
 					  stream, __sparse_cache);
-		comp_update_buffer_consume(src_c, mod->input_buffers[i].consumed);
+		if (!mod->skip_src_read_pointer_update)
+			comp_update_buffer_consume(src_c, mod->input_buffers[i].consumed);
```

Comment:
Is volume component not updating consumed correctly or not producing output_buffers[i].size when executing the ramp? Could this be fixed there instead?

Reply:
> Is volume component not updating consumed correctly or not producing output_buffers[i].size when executing the ramp? 

In our volume_process function, we just calculate the "size", but never update the read/write pointer of in/output stream. while executing ramp, we will execute the while loop several times, for example, each volume_process would process 48 frames, and each while loop( for ramp) process 6 frames, if we do not update the pointer in that while loop, we will get wrong data.

>Could this be fixed there instead?
I used to work around (record the initial pointers before while loop, and assign them to read/write pointers after while loop) and it works. I can figure out which performance better and use the best one?

Reply:
@andrula-song The description of this PR says: "we may need to run the while loop in volume_process function several times" whereas my understanding is that those loops shouldn't be run more than twice: normally they would run only once as long as there's no wrapping, and if data wrapped, the loop will run a second time to process the remaining part at the beginning of the buffer. Are you saying, that in your tests the loop ran more than twice? Was it with IPC3 or IPC4? That would only be possible if the input buffer were refilled while we were doing volume processing and that would only be possible if input were generated on a different core, but we don't support linked components on different cores. Could you explain your test case where this problem is observed?

Reply:
Volume as module adapter component should process input_buffers[0].size amount of samples and update ramp at the determined rate. If the block is 6 frames it would loop calculate 48 frames with 8 different ramp gains. The period # of frames is not necessarily an integer multiple of ramp gain block, so the last one can be different length.

Did you see issues with the fact that peakvolume process is called with input_buffers[0].size as zero a few times in the beginning of stream. All the ramp time counts should not start until the component receives data.

Reply:
It should be also possible to continue e.g. the ramp block of e.g. 6 across process periods to keep an even ramp update rate always. I'd go to that direction instead of adjusting the process amount to match exactly N * ramp block.

Reply:
"we may need to run the while loop in volume_process function several times"  for **ramp**. For example we process 48 frames every time we call the volume_process, while executing ramp(fade in),  we update gain per 6 frames(volume_ramp), we need to execute the while 8 times to finish the processing of 48 frames. But if the ramp is disabled, then the while loop run only once.

Reply:
Yes, and it's also very likely that ramp can end in the middle of process period. I think we've missed this regression when converting volume to module adapter (no tests done for ramp). I'd check first doing the changes to volume/peak volume process and keeping the component as normal simple copy client instead of adding new client mode to module adapter.

Reply:
> It should be also possible to continue e.g. the ramp block of e.g. 6 across process periods to keep an even ramp update rate always. I'd go to that direction instead of adjusting the process amount to match exactly N * ramp block.

yep, the exact number in my comment is just a example.